content/docs/next-release/sampling.md

```diff
@@ -83,3 +78,12 @@ In the above example:
 * All operations for service `bar` are rate-limited at 5 traces per second.
 * Any other service will be sampled with probability 0.5 defined by the `default_strategy`.
 * The `default_strategy` also includes shared per-operation strategies. In this example we disable tracing on `/health` and `/metrics` endpoints for all services by using probability 0. These per-operation strategies will apply to any new service not listed in the config, as well as to the `foo` and `bar` services unless they define their own strategies for these two operations.
+
+### Adaptive Sampling
+
+Adaptive sampler is a composite sampler that combines two functions:
+
+  * It makes sampling decisions on a per-operation basis, i.e. based on {{< tip "span" >}} operation name. This is especially useful in the API services whose endpoints may have very different traffic volumes and using a single probabilistic sampler for the whole service might starve (never sample) some of the low QPS endpoints.
+  * It supports a minimum guaranteed rate of sampling, such as always allowing up to N {{< tip "traces" "trace" >}} per seconds and then sampling anything above that with a certain probability (everything is per-operation, not per-service).
+
+Adaptive sampling requires a central store for the collectors to coordinate information about spans received/second and current sampling rates. Currently this has only been implemented for Cassandra and adaptive sampling shares the same storage as your span storage. Right now adaptive sampling will only work if `SPAN_STORAGE_TYPE` is set to Cassandra and the [relevant tables](https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/cassandra/schema/v003.cql.tmpl#L206-L229) have been created. If you're just getting started please review `sampling.initial-sampling-probability` and `sampling.target-samples-per-second`, and see the [cli parameters](/docs/latest/cli/#jaeger-collector-adaptive) for details on advanced configuration.
```

Comment:
Suggested text:

Adaptive sampling works in the Jaeger collector by observing the spans received from services and recalculating sampling probabilities for each service/endpoint combination to ensure that the volume of collected traces matches `--sampling.target-samples-per-second`. When a new service or endpoint is detected, it is initially sampled with `--sampling.initial-sampling-probability` until enough data is collected to calculate the rate appropriate for the traffic going through the endpoint.

Reply:
Once we merge this I would like to write a more detailed description of the actual algorithm.